packages/next/client/link.tsx

```diff
@@ -125,11 +117,6 @@ function linkClicked(
     return
   }
 
-  if (!isLocal(href)) {
```

Comment:
Let's move this check up into the only when `<a>` clause above and keep it. This will prevent us from hijacking the default link behavior.